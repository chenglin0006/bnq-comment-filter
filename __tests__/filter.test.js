import {shallow} from 'enzyme';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Filter from '../src/index'
import renderer from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });

import {Form} from "antd/lib/index";
const WrappedAdvancedFilter = Form.create()(Filter);

test('filter', () => {
    const props = {
        // Jest 提供的mock 函数
        filterData:[{
            id: 'activityCode',
            name: '活动编码',
            inputChangeName:'titleChangeFun'
        },{
            id: 'searchParam',
            name: '查询条件',
        }],
        handleReset: jest.fn( (e) => {
            console.log('reset','=============')
        }),
        handleSearch:jest.fn( (e) => {
            console.log('search','=============')
        }),
        handleBack:jest.fn( (e) => {
            console.log('back','=============')
        }),
    }
    const filterWrapperTest = shallow(<div className={'test'} onClick={()=>{
        console.log('test');
    }}><WrappedAdvancedFilter {...props} ></WrappedAdvancedFilter></div>);
    filterWrapperTest.find('div.test').simulate('click');
    filterWrapperTest.find('button.reset').simulate('click');
});
