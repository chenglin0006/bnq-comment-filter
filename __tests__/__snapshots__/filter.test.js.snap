// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter simulates click events 1`] = `
<div
  className="bnq-common-filter-container"
>
  <form
    className="ant-form ant-form-inline filter-form"
    onSubmit={[Function]}
  >
    <div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="ant-form-item-label"
        >
          <label
            className=""
            htmlFor="activityCode"
            onClick={[Function]}
            title="活动编码"
          >
            活动编码
          </label>
        </div>
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                data-__field={
                  Object {
                    "name": "activityCode",
                  }
                }
                data-__meta={
                  Object {
                    "initialValue": undefined,
                    "name": "activityCode",
                    "originalProps": Object {
                      "disabled": false,
                      "placeholder": undefined,
                      "prefixCls": "ant-input",
                      "type": "text",
                    },
                    "ref": null,
                    "trigger": "onChange",
                    "validate": Array [],
                    "valuePropName": "value",
                  }
                }
                disabled={false}
                id="activityCode"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="ant-form-item-label"
        >
          <label
            className=""
            htmlFor="searchParam"
            onClick={[Function]}
            title="查询条件"
          >
            查询条件
          </label>
        </div>
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                data-__field={
                  Object {
                    "name": "searchParam",
                  }
                }
                data-__meta={
                  Object {
                    "initialValue": undefined,
                    "name": "searchParam",
                    "originalProps": Object {
                      "disabled": false,
                      "placeholder": undefined,
                      "prefixCls": "ant-input",
                      "type": "text",
                    },
                    "ref": null,
                    "trigger": "onChange",
                    "validate": Array [],
                    "valuePropName": "value",
                  }
                }
                disabled={false}
                id="searchParam"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="btnContainer filter-action-div"
    >
      <button
        className="ant-btn search ant-btn-primary"
        onClick={[Function]}
        type="submit"
      >
        <span>
          搜 索
        </span>
      </button>
      <button
        className="ant-btn reset"
        onClick={[Function]}
        type="button"
      >
        <span>
          重 置
        </span>
      </button>
      <button
        className="ant-btn back"
        onClick={[Function]}
        type="button"
      >
        <span>
          返 回
        </span>
      </button>
      
    </div>
    <div
      className="ant-divider ant-divider-horizontal"
    />
  </form>
</div>
`;
